src/tab.rs

```diff
@@ -199,6 +202,60 @@ fn format_size(size: u64) -> String {
         format!("{} B", size)
     }
 }
+enum PermissionOwner {
+    Owner,
+    Group,
+    Other,
+}
+
+fn format_permissions_owner(metadata: &Metadata, owner: PermissionOwner) -> String {
+    return match owner {
+        PermissionOwner::Owner => get_user_by_uid(metadata.uid())
+            .and_then(|user| user.name().to_str().map(ToOwned::to_owned))
+            .unwrap_or_default(),
+        PermissionOwner::Group => get_group_by_gid(metadata.gid())
+            .and_then(|group| group.name().to_str().map(ToOwned::to_owned))
+            .unwrap_or_default(),
+        PermissionOwner::Other => String::from(""),
+    };
+}
+fn format_permissions(metadata: &Metadata, owner: PermissionOwner) -> String {
+    let readable = match owner {
+        PermissionOwner::Owner => metadata.permissions().readable_by_owner(),
+        PermissionOwner::Group => metadata.permissions().readable_by_group(),
+        PermissionOwner::Other => metadata.permissions().readable_by_other(),
+    };
+    let writeable = match owner {
+        PermissionOwner::Owner => metadata.permissions().writable_by_owner(),
+        PermissionOwner::Group => metadata.permissions().writable_by_group(),
+        PermissionOwner::Other => metadata.permissions().writable_by_other(),
+    };
+    let executable = match owner {
+        PermissionOwner::Owner => metadata.permissions().executable_by_owner(),
+        PermissionOwner::Group => metadata.permissions().executable_by_group(),
+        PermissionOwner::Other => metadata.permissions().executable_by_other(),
+    };
+    format!(
+        "{} {} {}",
```

Comment:
I believe if this is readable and executable but not writable, there will be an extra space between Read and Execute.

Reply:
One way to fix this would be to push the read, write, and execute strings to a Vec and use join(' ') to turn it into a string.

Reply:
Thanks for the review! Bug fixed, and the code is _much_ cleaner now.